mdata/cache/ccache.go

```diff
@@ -58,6 +58,31 @@ func (c *CCache) evictLoop() {
 	}
 }
 
+// adds the given chunk to the cache, but only if the metric is sufficiently hot
+func (c *CCache) CacheIfHot(metric string, prev uint32, itergen *chunk.IterGen) {
+	c.RLock()
+
+	var met *CCacheMetric
+	var ok bool
+
+	// if this metric is not cached at all it is not hot
+	if met, ok = c.metricCache[metric]; !ok {
+		c.RUnlock()
+		return
+	}
+
+	// if the previous chunk is not cached we consider the metric not hot enough to cache this chunk
+	if met.lastTs() < itergen.Ts() {
```

Comment:
this only works reliably for span aware chunks right? (which would be OK) but maybe worth mentioning.